functions/go/upsert-resource/upsertresource/upsert_resource.go

```diff
@@ -8,21 +8,62 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// UpsertResource upserts resource to input list of resources
-// resources are uniquely identifies by Group, Kind, Name and Namespace
+// UpsertResource upserts resources to input list of resources
+// resources are uniquely identifies by Group, Kind, Name, Namespace and input file path
 type UpsertResource struct {
-	// Resource is the input resource for upsert
-	Resource *yaml.RNode
+	// UpsertResourceList input resources to upsert
+	UpsertResourceList *yaml.RNode
 }
 
+const (
+	DestPathAnnotation     = "config.kubernetes.io/upsert-path"
```

Comment:
hmm...I think it will be good to narrow the scope for annotations to this function. `config.kubernetes.io` is generic prefix being used in k8s ecosystem. @mengqiy any recommendation ?

Reply:
I don't think this annotation should be named specifically for `upsert`, since if/when we add other functions for modifying resources directly (ex. `patch`) I'd expect them to use the same annotation.

Maybe `target-path`?

Reply:
in the current design, annotation `DestPathAnnotation` has the same meaning as `internal.config.kubernetes.io/path`. IMO I'd suggest using an annotation with kpt prefix, so users a less likely to confuse these 2 annotations.
Maybe we can consider something like `config.kpt-dev/target-path` or `fn.kpt-dev/target-path`.

Reply:
we are going with config.kubernetes.io/target-path